mono/mini/mini-profiler.c

```diff
@@ -49,39 +50,49 @@ emit_fill_call_ctx (MonoCompile *cfg, MonoInst *method, MonoInst *ret)
 void
 mini_profiler_emit_enter (MonoCompile *cfg)
 {
-	if (!MONO_CFG_PROFILE (cfg, ENTER) || cfg->current_method != cfg->method)
+	gboolean trace = mono_jit_trace_calls != NULL && mono_trace_eval (cfg->method);
```

Comment:
Perhaps we should store the result of `mono_trace_eval` as a bit on the CFG so we don't have to call it both here and in `mini_method_compile`?